File: tyama_icpc2007rA.go
//usr/bin/env go run $0 $@;exit
package main
import(
	"fmt"
	"os"
	"text/scanner"
	"strconv"
)

var sin scanner.Scanner
func scanint() int{
	sin.Scan()
	ret,_ := strconv.Atoi(sin.TokenText())
	return ret
}
func main(){
	sin.Init(os.Stdin)
	for {
		n:=scanint()
		k:=scanint()
		m:=scanint()
		if n==0 {break}
		r:=0
		for i:=1;i<n;i++ {r=(r+k)%i}
		fmt.Printf("%d\n",(r+m)%n+1)
	}
}

File: tyama_icpc2007rA.R
#!/usr/bin/env Rscript
x=scan("stdin")
j=0
repeat{
	n=x[j+1]
	k=x[j+2]
	m=x[j+3]
	if(n==0)break
	r=0
	for(i in 1:(n-1))r=(r+k)%%i
	cat((r+m)%%n+1)
	cat("\n")
	j=j+3
}